nutanix_volume_group_disk_v2 encounters panic when updating disk_size_bytes

### Nutanix Cluster Information

- PC 2024.3
- AOS 7.0

### Terraform Version

- 1.12.2

### Affected Resource(s)

- nutanix_volume_group_disk_v2

### Terraform Configuration Files
```hcl
locals {
  cluster_extId = try(
    [for cluster in data.nutanix_clusters.clusters.entities : cluster.metadata.uuid if cluster.name == var.cluster_name][0],
    [for cluster in data.nutanix_clusters.clusters.entities : cluster.metadata.uuid if cluster.service_list[0] != "PRISM_CENTRAL"][0]
  )
}

data "nutanix_clusters" "clusters" {}

data "nutanix_storage_containers_v2" "storage_container_ext_id" {
  filter = "clusterExtId  eq '${local.cluster_extId}' and name eq '${var.storage_container_name}'"
}

resource "random_pet" "vg-name" {}

resource "nutanix_volume_group_v2" "volume_group" {
  name                               = "vg-${random_pet.vg-name.id}"
  should_load_balance_vm_attachments = false
  sharing_status                     = "SHARED"

  cluster_reference = local.cluster_extId

  is_hidden = false

}

resource "nutanix_volume_group_disk_v2" "volume_group_disk" {

  volume_group_ext_id = nutanix_volume_group_v2.volume_group.ext_id
  index               = 1
  description         = "FIRST_DISK"
  disk_size_bytes     = provider::units::from_gb(2)

  disk_data_source_reference {
    ext_id      = data.nutanix_storage_containers_v2.storage_container_ext_id.storage_containers[0].ext_id
    entity_type = "STORAGE_CONTAINER"
  }

  lifecycle {
    ignore_changes = [
      disk_data_source_reference
    ]
  }

}
```
### Debug Output
<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
### Panic Output
```
│ Error: Plugin did not respond
│ 
│   with nutanix_volume_group_disk_v2.volume_group_disk,
│   on main.tf line 95, in resource "nutanix_volume_group_disk_v2" "volume_group_disk":
│   95: resource "nutanix_volume_group_disk_v2" "volume_group_disk" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-nutanix_v2.2.1 plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 83 [running]:
github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/volumesv2.ResourceNutanixVolumeGroupDiskV2Update({0x105672ed8, 0x1400141df10}, 0x14001451080, {0x105423f40?, 0x14000c79cb0?})
        github.com/terraform-providers/terraform-provider-nutanix/nutanix/services/volumesv2/resource_nutanix_volume_group_disk_v2.go:233 +0xb78
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x14000a8b180, {0x105672e68, 0x14001412e60}, 0x24?, {0x105423f40, 0x14000c79cb0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:375 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000a8b180, {0x105672e68, 0x14001412e60}, 0x14001442680, 0x14001450f00, {0x105423f40, 0x14000c79cb0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:475 +0x4d4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000234078, {0x105672e68, 0x14001412e60}, 0x14001412eb0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xabc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000278380, {0x105672e30?, 0x14001446b40?}, 0x1400141cd90)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x20c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x105615f20?, 0x14000278380}, {0x105672e30, 0x14001446b40}, 0x140013b2ea0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002c5500, {0x1056778f0, 0x140001d6780}, 0x140001ba200, 0x140009ff290, 0x105e0eea0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0xa58
google.golang.org/grpc.(*Server).handleStream(0x140002c5500, {0x1056778f0, 0x140001d6780}, 0x140001ba200, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0x7cc
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.32.0/server.go:859 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 21
        google.golang.org/grpc@v1.32.0/server.go:857 +0x24c

Error: The terraform-provider-nutanix_v2.2.1 plugin crashed!
```
### Expected Behavior
Volume Group disk resource size is updated
### Actual Behavior
Provider panic and disk size fails to update
### Steps to Reproduce
1. `terraform apply`

--- a/nutanix/services/volumesv2/data_source_nutanix_volume_disk_v2_test.go
+++ b/nutanix/services/volumesv2/data_source_nutanix_volume_disk_v2_test.go
@@ -23,7 +23,7 @@ func TestAccV2NutanixVolumeGroupsDiskDataSource_Basic(t *testing.T) {
 		Providers: acc.TestAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccVolumeGroupsDiskDataSourceConfig(filepath, name, desc),
+				Config: testAccVolumeGroupsDiskDataSourceConfig(filepath, name, desc, int(diskSizeBytes)),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(dataSourceVolumeGroupsDisk, "index", "1"),
 					resource.TestCheckResourceAttr(dataSourceVolumeGroupsDisk, "disk_size_bytes", strconv.Itoa(int(diskSizeBytes))),
@@ -35,9 +35,9 @@ func TestAccV2NutanixVolumeGroupsDiskDataSource_Basic(t *testing.T) {
 	})
 }
 
-func testAccVolumeGroupsDiskDataSourceConfig(filepath, name, desc string) string {
+func testAccVolumeGroupsDiskDataSourceConfig(filepath, name, desc string, diskSizeBytes int) string {
 	return testAccVolumeGroupResourceConfig(name, desc) +
-		testAccVolumeGroupDiskResourceConfig(name, desc) +
+		testAccVolumeGroupDiskResourceConfig(name, desc, diskSizeBytes) +
 		`		  
 		  data "nutanix_volume_group_disk_v2" "test" {
 			volume_group_ext_id = nutanix_volume_group_v2.test.id

--- a/nutanix/services/volumesv2/data_source_nutanix_volume_disks_v2_test.go
+++ b/nutanix/services/volumesv2/data_source_nutanix_volume_disks_v2_test.go
@@ -23,7 +23,7 @@ func TestAccV2NutanixVolumeGroupsDisksDataSource_Basic(t *testing.T) {
 		Providers: acc.TestAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccVolumeGroupsDisksDataSourceConfig(filepath, name, desc),
+				Config: testAccVolumeGroupsDisksDataSourceConfig(filepath, name, desc, int(diskSizeBytes)),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckResourceAttrListNotEmpty(dataSourceVolumeGroupsDisks, "disks", "index"),
 					resource.TestCheckResourceAttrSet(dataSourceVolumeGroupsDisks, "disks.#"),
@@ -48,7 +48,7 @@ func TestAccV2NutanixVolumeGroupsDisksDataSource_WithLimit(t *testing.T) {
 		Providers: acc.TestAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccVolumeGroupsDisksDataSourceWithLimit(filepath, name, desc, limit),
+				Config: testAccVolumeGroupsDisksDataSourceWithLimit(filepath, name, desc, int(diskSizeBytes), limit),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckResourceAttrListNotEmpty(dataSourceVolumeGroupsDisks, "disks", "index"),
 					resource.TestCheckResourceAttr(dataSourceVolumeGroupsDisks, "disks.#", "1"),
@@ -80,8 +80,8 @@ func TestAccV2NutanixVolumeGroupsDisksDataSource_WithInvalidFilter(t *testing.T)
 	})
 }
 
-func testAccVolumeGroupsDisksDataSourceConfig(filepath, name, desc string) string {
-	return testAccVolumeGroupResourceConfig(name, desc) + testAccVolumeGroupDiskResourceConfig(name, desc) +
+func testAccVolumeGroupsDisksDataSourceConfig(filepath, name, desc string, diskSizeBytes int) string {
+	return testAccVolumeGroupResourceConfig(name, desc) + testAccVolumeGroupDiskResourceConfig(name, desc, diskSizeBytes) +
 		fmt.Sprintf(`
 
 
@@ -116,8 +116,8 @@ func testAccVolumeGroupsDisksDataSourceConfig(filepath, name, desc string) strin
 	`, desc, diskSizeBytes)
 }
 
-func testAccVolumeGroupsDisksDataSourceWithLimit(filepath, name, desc string, limit int) string {
-	return testAccVolumeGroupResourceConfig(name, desc) + testAccVolumeGroupDiskResourceConfig(name, desc) +
+func testAccVolumeGroupsDisksDataSourceWithLimit(filepath, name, desc string, diskSizeBytes int, limit int) string {
+	return testAccVolumeGroupResourceConfig(name, desc) + testAccVolumeGroupDiskResourceConfig(name, desc, diskSizeBytes) +
 		fmt.Sprintf(`
 
 	  	resource "nutanix_volume_group_disk_v2" "test-2" {

--- a/nutanix/services/volumesv2/helpers_test.go
+++ b/nutanix/services/volumesv2/helpers_test.go
@@ -15,6 +15,7 @@ import (
 )
 
 var diskSizeBytes int64 = 5368709120
+var updatedDiskSizeBytes int64 = 10737418240
 
 func testAccCheckResourceAttrListNotEmpty(resourceName, attrName, subAttr string) resource.TestCheckFunc {
 	log.Printf("[DEBUG] testAccCheckResourceAttrNotEmpty ###############################")
@@ -137,7 +138,7 @@ func testAccVolumeGroupResourceConfig(name, desc string) string {
 	`, filepath, name, desc)
 }
 
-func testAccVolumeGroupDiskResourceConfig(name, desc string) string {
+func testAccVolumeGroupDiskResourceConfig(name, desc string, diskSizeBytes int) string {
 	return fmt.Sprintf(`	  
 
       data "nutanix_storage_containers_v2" "test" {

--- a/nutanix/services/volumesv2/resource_nutanix_volume_group_disk_v2.go
+++ b/nutanix/services/volumesv2/resource_nutanix_volume_group_disk_v2.go
@@ -230,7 +230,7 @@ func ResourceNutanixVolumeGroupDiskV2Update(ctx context.Context, d *schema.Resou
 		updateSpec.Index = nil
 	}
 	if d.HasChange("disk_size_bytes") {
-		diskSizeBytes := d.Get("disk_size_bytes").(int64)
+		diskSizeBytes := int64(d.Get("disk_size_bytes").(int))
 		updateSpec.DiskSizeBytes = &diskSizeBytes
 	}
 	if d.HasChange("description") {

--- a/nutanix/services/volumesv2/resource_nutanix_volume_group_disk_v2_test.go
+++ b/nutanix/services/volumesv2/resource_nutanix_volume_group_disk_v2_test.go
@@ -22,7 +22,7 @@ func TestAccV2NutanixVolumeGroupDiskResource_Basic(t *testing.T) {
 		Providers: acc.TestAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccVolumeGroupsDiskResourceConfig(filepath, name, desc),
+				Config: testAccVolumeGroupsDiskResourceConfig(filepath, name, desc, int(diskSizeBytes)),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "description", desc),
 					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "disk_size_bytes", strconv.Itoa(int(diskSizeBytes))),
@@ -34,7 +34,37 @@ func TestAccV2NutanixVolumeGroupDiskResource_Basic(t *testing.T) {
 	})
 }
 
-func testAccVolumeGroupsDiskResourceConfig(filepath, name, desc string) string {
+func TestAccV2NutanixVolumeGroupDiskResource_BasicUpdateDiskSize(t *testing.T) {
+	r := acctest.RandInt()
+	name := fmt.Sprintf("tf-test-volume-group-%d", r)
+	desc := "test volume group disk description"
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { acc.TestAccPreCheck(t) },
+		Providers: acc.TestAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVolumeGroupsDiskResourceConfig(filepath, name, desc, int(diskSizeBytes)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "description", desc),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "disk_size_bytes", strconv.Itoa(int(diskSizeBytes))),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "index", "1"),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "disk_storage_features.0.flash_mode.0.is_enabled", "false"),
+				),
+			},
+			{
+				Config: testAccVolumeGroupsDiskResourceConfig(filepath, name, desc, int(updatedDiskSizeBytes)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "description", desc),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "disk_size_bytes", strconv.Itoa(int(updatedDiskSizeBytes))),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "index", "1"),
+					resource.TestCheckResourceAttr(resourceVolumeGroupDisk, "disk_storage_features.0.flash_mode.0.is_enabled", "false"),
+				),
+			},
+		},
+	})
+}
+
+func testAccVolumeGroupsDiskResourceConfig(filepath, name, desc string, diskSizeBytes int) string {
 	return testAccVolumeGroupResourceConfig(name, desc) +
-		testAccVolumeGroupDiskResourceConfig(name, desc)
+		testAccVolumeGroupDiskResourceConfig(name, desc, diskSizeBytes)
 }

